src/Executables/Game/GuildManager.cs

```diff
@@ -0,0 +1,245 @@
+using EnumsNET;
+using Microsoft.EntityFrameworkCore;
+using QuantumCore.API;
+using QuantumCore.API.Game.Guild;
+using QuantumCore.Game.Persistence;
+using QuantumCore.Game.Persistence.Entities.Guilds;
+using QuantumCore.Game.Persistence.Extensions;
+using GuildMember = QuantumCore.Game.Persistence.Entities.Guilds.GuildMember;
+
+namespace QuantumCore.Game;
+
+public class GuildManager : IGuildManager
+{
+    private readonly GameDbContext _db;
+    private readonly IGuildExperienceManager _experienceManager;
+
+    public GuildManager(GameDbContext db, IGuildExperienceManager experienceManager)
+    {
+        _db = db;
+        _experienceManager = experienceManager;
+    }
+
+    public async Task<GuildData?> GetGuildByNameAsync(string name, CancellationToken token = default)
+    {
+        return await _db.Guilds
+            .Where(x => x.Name == name)
+            .SelectData()
+            .FirstOrDefaultAsync(token);
+    }
+
+    public async Task<GuildData?> GetGuildByIdAsync(uint id, CancellationToken token)
+    {
+        return await _db.Guilds
+            .Where(x => x.Id == id)
+            .SelectData()
+            .FirstOrDefaultAsync(token);
+    }
+
+    public async Task<GuildData?> GetGuildForPlayerAsync(uint playerId, CancellationToken token = default)
+    {
+        return await _db.Guilds
+            .Where(x => x.Members.Any(m => m.PlayerId == playerId))
+            .SelectData()
+            .FirstOrDefaultAsync(token);
+    }
+
+    public async Task<GuildData> CreateGuildAsync(string name, uint leaderId, CancellationToken token = default)
+    {
+        var leaderRank = new GuildRank
+        {
+            Position = 1,
+            Name = "Leader",
+            Permissions = GuildRankPermission.All
+        };
+        var guild = new Guild
+        {
+            Name = name,
+            OwnerId = leaderId,
+            Level = 1,
+            Members = new List<GuildMember>
+            {
+                new GuildMember
+                {
+                    PlayerId = leaderId,
+                    IsLeader = true,
+                    Rank = leaderRank
+                }
+            },
+            Ranks = Enumerable.Range(2, GuildConstants.RANKS_LENGTH - 1)
+                .Select(rank => new GuildRank
+                {
+                    Position = (byte) rank,
+                    Name = "Member"
+                })
+                .Prepend(leaderRank)
+                .ToList(),
+            MaxMemberCount = GuildConstants.MEMBERS_MAX_DEFAULT
+        };
+        _db.Guilds.Add(guild);
+        await _db.SaveChangesAsync(token);
+        await _db.Players
+            .Where(x => x.Id == leaderId)
+            .ExecuteUpdateAsync(setter => setter
+                .SetProperty(p => p.GuildId, guild.Id), token);
+
+        return new GuildData
+        {
+            Id = guild.Id,
+            Name = name,
+            Level = 1,
+            MaxMemberCount = 10,
```

Comment:
Magic number 👀

also, why don't we get the `GuildData.MaxMemberCount` from `guild.MaxMemberCount`?

Reply:
That's a bug. Well spotted 👍